nym-vpn-x/src-tauri/src/grpc/client.rs

```diff
@@ -22,37 +36,61 @@ pub enum VpndStatus {
 }
 
 #[derive(Debug, Default, Clone)]
-pub struct GrpcClient {
-    pub endpoint: String,
-}
+pub struct GrpcClient(Transport);
 
 impl GrpcClient {
-    pub fn new(address: &str) -> Self {
-        Self {
-            endpoint: address.to_string(),
+    #[instrument(skip_all)]
+    pub fn new(config: &AppConfig, cli: &Cli) -> Self {
+        let client = GrpcClient(Transport::from((config, cli)));
+        match &client.0 {
+            Transport::Http(endpoint) => {
+                info!("using grpc HTTP transport: {}", endpoint);
+            }
+            Transport::Ipc(socket) => {
+                info!("using grpc IPC transport: {}", socket.display());
+            }
         }
+        client
     }
 
     /// Get the Vpnd service client
     #[instrument(skip_all)]
     pub async fn vpnd(&self) -> Result<NymVpndClient<Channel>> {
-        NymVpndClient::connect(self.endpoint.clone())
-            .await
-            .inspect_err(|e| {
-                warn!("failed to connect to the daemon: {:?}", e);
-            })
-            .map_err(|e| anyhow!("failed to connect to the daemon: {}", e))
+        match &self.0 {
+            Transport::Http(endpoint) => {
+                NymVpndClient::connect(endpoint.clone()).await.map_err(|e| {
+                    warn!("failed to connect to the daemon: {}", e);
+                    anyhow!("failed to connect to the daemon: {}", e)
+                })
+            }
+            Transport::Ipc(socket) => {
+                let channel = get_channel(socket.clone()).await.map_err(|e| {
+                    warn!("failed to connect to the daemon: {}", e);
+                    anyhow!("failed to connect to the daemon: {}", e)
```

Comment:
This pattern of both having a log warning and return an error for the same thing seems like a code smell. Log these things at the top-level when you need to

This comment applies to all of them

Reply:
The above comment is primarily for future PRs, and using anyhow here seems like a quick hack for now until we setup error enums for these components. So yeah something to keep that in mind for follow up PRs